server/utils/azureAppInsights.test.ts

```diff
@@ -0,0 +1,89 @@
+import { EnvelopeTelemetry } from 'applicationinsights/out/Declarations/Contracts'
+import { ClientRequest } from 'http'
+import { addUserDataToRequests } from './azureAppInsights'
+
+const user = {
+  activeCaseLoadId: 'LII',
+  isReceptionUser: true,
+  username: 'test-user',
+  activeCaseLoadDescription: 'Lincoln',
+}
+
+const createEnvelope = (properties: Record<string, string | boolean>, baseType = 'RequestData') =>
+  ({
+    data: {
+      baseType,
+      baseData: { properties },
+    },
+  } as unknown as EnvelopeTelemetry)
```

Comment:
the as unknown as <TYPE> feels a bit hacky, anyone know a better way to quickly define a type without having to specify the entire structure?

Reply:
it usually works without `as unknown` if it contains a subset of the object. This suggests maybe the typing is a bit off, things in the wrong place?